### DISCUSSION TITLE
sensor layout unrecognized when running ANOVA

### DISCUSSION THREAD

**Discussion by giadaan** [Eelbrain/Eelbrain#68] [Q&A]
Hello all,

yesterday I tried to run an ANOVA that I had previously set in the experiment class. In particular, I ran:

<e.load_test('anova',group='cy',make=True,data='sensor')>

and got the following output error:

{
	"name": "ValueError",
	"message": "data='planar1'",
	"stack": "---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/export/home/gantonicelli/public/PROTRACK_MEG/Code/mTRF_eelbrain.ipynb Cell 5 line 1
----> <a href='vscode-notebook-cell:/export/home/gantonicelli/public/PROTRACK_MEG/Code/mTRF_eelbrain.ipynb#W4sZmlsZQ%3D%3D?line=0'>1</a> e.load_test('anova',group='cy',make=True,data='sensor')

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_experiment/mne_experiment.py:3856, in MneExperiment.load_test(self, test, tstart, tstop, pmin, parc, mask, samples, data, baseline, smooth, src_baseline, return_data, make, **state)
   3854 data = TestDims.coerce(data, morph=True)
   3855 self._set_analysis_options(data, baseline, src_baseline, pmin, tstart, tstop, parc, mask, smooth=smooth)
-> 3856 return self._load_test(test, tstart, tstop, pmin, parc, mask, samples, data, baseline, src_baseline, return_data, make, smooth)

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_experiment/mne_experiment.py:3944, in MneExperiment._load_test(self, test, tstart, tstop, pmin, parc, mask, samples, data, baseline, src_baseline, return_data, make, smooth)
   3942 else:
   3943     if data.sensor:
-> 3944         res_data = self.load_evoked(True, baseline, True, test_obj.cat, data=data, vardef=test_obj.vars)
   3945     elif data.source:
   3946         res_data = self.load_evoked_stc(True, baseline, src_baseline, morph=True, cat=test_obj.cat, mask=mask, vardef=test_obj.vars)

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_experiment/mne_experiment.py:2768, in MneExperiment.load_evoked(self, subjects, baseline, ndvar, cat, samplingrate, decim, data_raw, vardef, data, **state)
   2766 connectivity = pipe.get_connectivity(data_kind)
   2767 name = 'meg' if data_kind == 'mag' else data_kind
-> 2768 ds[name] = load.mne.evoked_ndvar(evoked, data=data_kind, sysname=sysname, connectivity=connectivity)
   2769 if data_kind != 'eog' and isinstance(data.sensor, str):
   2770     ds[name] = getattr(ds[name], data.sensor)('sensor')

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_io/fiff.py:1269, in evoked_ndvar(evoked, name, data, exclude, vmax, sysname, connectivity)
   1267     info = _info.for_meg(vmax, unit='T/cm')
   1268 else:
-> 1269     raise ValueError(\"data=%s\" % repr(data))
   1271 e0 = evoked[0]
   1272 if len(evoked) == 1:

ValueError: data='planar1'"
}

Any help would be very much appreciated :)

**Comment by christianbrodbeck** [comment]
You are trying to load a mass-univariate test in sensor space for a MEG system that has more than one sensor type. This is currently not implemented. You can restrict the sensor type by using e.g. `data='mag'`.

More generally, mass-univariate tests *could* be extended to multiple sensor types but I think it would require some thought also on theoretical issues (how would you expect clusters to form? Should clusters always be restricted to a single sensor type? Or should they be purely spatial, i.e. be allowed to cross over between different sensor types? Do different sensor types have different SNR and could that hurt the test?). If someone is interested in that I'd be happy to talk more.

The difficult-to-interpret error is due to lack of testing with your MEG system, I will improve the error message.

**Reply by giadaan** [reply]
When setting data='meg' it gives me no error, while data='mag' is not allowed apparently.

**Reply by christianbrodbeck** [reply]
`mag`, what is there error? Are you using version 0.40a7 ?

**Reply by giadaan** [reply]
version 0.39.8

**Reply by christianbrodbeck** [reply]
Please always try the latest alpha (currently 0.40a7) first, since the MneExperiment has not been developed for your MEG system I can only improve support for it in the alpha version.
